src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfTTypeToConvert.cs

```diff
@@ -139,8 +139,8 @@ public override bool ReadJsonAndSetMember(object obj, ref ReadStack state, ref U
             {
                 if (!IgnoreNullValues)
                 {
-                    TTypeToConvert value = default!;
-                    Set!(obj, value);
+                    TTypeToConvert value = default;
+                    Set!(obj, value!);
```

Comment:
@jcouv this one is weird. Why can't I pass a possible null value to `Action<object, TTypeToConvert>`? Isn't supposed that that the input Generic values are treated as nullable by default?

Reply:
Consider if `TTypeToConvert` were a non-nullable type like `object!`. 
You don't want to pass a `null` value to the second parameter of `Action<object, object>`.

---
In reply to: [381017763](https://github.com/dotnet/runtime/pull/32474#discussion_r381017763) [](ancestors = 381017763)

Reply:
FYI, I'm OOF soon, in case you have more questions. Check my OOF message for details.

---
In reply to: [381036451](https://github.com/dotnet/runtime/pull/32474#discussion_r381036451) [](ancestors = 381036451,381017763)